src/metadata_validator.py

```diff
@@ -149,8 +149,41 @@ def validateNode(self, dataRecord, model):
         
         return STATUS_PASSED, None, None
     
-    def validate_required_props(self, dataRecord, model):
-        return None
+    def validate_required_props(self, data_record, node_definition):
+        result = {"result": ERROR, "errors": [], "warnings": []}
+        # check the correct format from the node_definition
+        if "model" not in node_definition.keys() or "nodes" not in node_definition["model"].keys():
+            result["errors"].append(f"node definition is not correctly formatted.")
+            return result
+        # check the correct format from the data_record
+        if "nodeType" not in data_record.keys() or "rawData" not in data_record.keys() or len(data_record["rawData"].items()) == 0:
+            result["errors"].append(f"data record is not correctly formatted.")
+            return result
+
+        # validation start
+        nodes = node_definition["model"]["nodes"]
+        node_type = data_record["nodeType"]
+        # extract a node from the data record
+        if node_type not in nodes.keys():
+            result["errors"].append(f"Required node '{node_type}' does not exist.")
+            return result
+
+        anode = nodes[node_type]
+        if node_type in nodes.keys():
+            for key, value in data_record["rawData"].items():
+                anode_keys = anode.keys()
+                if "properties" not in anode_keys:
+                    result["errors"].append(f"data record is not correctly formatted.")
+                    break
+
+                key_not_exist = key not in anode["properties"]
+                value_invalid = not value.strip()
```

Comment:
do yo need to check if the prop is required?

Reply:
The primary goal of this PR is to check whether the necessary properties in the data_record are present and not empty. This is required.